lib/puppet_fixtures.rb

```diff
@@ -65,67 +65,66 @@ def symlinks
-                                                   {}
-                                                 end
-                      categories['forge_modules'] ||= {}
-                      categories['repositories'] ||= {}
-
-                      defaults = { 'target' => module_target_dir }
-
-                      ['symlinks', 'forge_modules', 'repositories'].to_h do |category|
-                        # load defaults from the `.fixtures.yml` `defaults` section
-                        # for the requested category and merge them into my defaults
-                        if (category_defaults = categories.dig('defaults', category))
-                          category_defaults = defaults.merge(category_defaults)
-                        else
-                          category_defaults = defaults
-                        end
-
-                        entries = categories[category].to_h do |fixture, opts|
-                          # convert a simple string fixture to a hash, by
-                          # using the string fixture as the `repo` option of the hash.
-                          if opts.instance_of?(String)
-                            opts = { 'repo' => opts }
-                          end
-                          # there should be a warning or something if it's not a hash...
-                          next unless opts.instance_of?(Hash)
-
-                          # merge our options into the defaults to get the
-                          # final option list
-                          opts = category_defaults.merge(opts)
-
-                          next unless include_repo?(opts['puppet_version'])
-
-                          entry = validate_fixture_hash!(
-                            target: File.join(opts['target'], fixture),
-                            ref: opts['ref'] || opts['tag'],
-                            branch: opts['branch'],
-                            scm: opts.fetch('scm', 'git'),
-                            flags: opts['flags'],
-                            subdir: opts['subdir'],
-                          )
-
-                          case category
-                          when 'forge_modules'
-                            entry.delete(:scm)
-                            entry.delete(:branch)
-                            entry.delete(:subdir)
-                          when 'symlinks'
-                            entry = PuppetFixtures::Symlink.new(link: entry[:target], target: opts['repo'])
-                          end
-
-                          [opts['repo'], entry]
-                        end
-
-                        [category, entries]
-                      end
-                    end
+        categories = read_fixtures_file['fixtures']
+
+        categories['symlinks'] ||= begin
+          metadata = PuppetFixtures::Metadata.new(File.join(source_dir,
+                                                            'metadata.json'))
```

Comment:
I don't think this is more readable
```suggestion
          metadata = PuppetFixtures::Metadata.new(File.join(source_dir, 'metadata.json'))
```